scipy/stats/_stats_py.py

```diff
@@ -540,41 +540,39 @@ def _mask_to_limits(a, limits, inclusive):
     limits : (float or None, float or None)
         A tuple consisting of the (lower limit, upper limit).  Values in the
         input array less than the lower limit or greater than the upper limit
-        will be masked out. None implies no limit.
+        will be replaced with `np.nan`. None implies no limit.
     inclusive : (bool, bool)
         A tuple consisting of the (lower flag, upper flag).  These flags
         determine whether values exactly equal to lower or upper are allowed.
 
-    Returns
-    -------
-    A MaskedArray.
-
-    Raises
-    ------
-    A ValueError if there are no values within the given limits.
-
     """
+    if limits is None:
+        return a
+    a = a.copy()
+    mask = np.full_like(a, False, dtype=np.bool_)
     lower_limit, upper_limit = limits
     lower_include, upper_include = inclusive
-    am = ma.MaskedArray(a)
     if lower_limit is not None:
-        if lower_include:
-            am = ma.masked_less(am, lower_limit)
-        else:
-            am = ma.masked_less_equal(am, lower_limit)
-
+        mask |= a < lower_limit
+        if not lower_include:
+            mask |= a == lower_limit
     if upper_limit is not None:
-        if upper_include:
-            am = ma.masked_greater(am, upper_limit)
-        else:
-            am = ma.masked_greater_equal(am, upper_limit)
-
-    if am.count() == 0:
+        mask |= a > upper_limit
+        if not upper_include:
+            mask |= a == upper_limit
+    if np.all(mask):
         raise ValueError("No array values within given limits")
-
-    return am
+    if np.any(mask):
+        if not np.issubdtype(a.dtype, np.inexact):
+            a = a.astype(np.float64)
+        np.place(a, mask, np.nan)
```

Comment:
100% subjective nit: if you don't care either way, consider:
```suggestion
        a[mask] = np.nan
```

Neither this nor `place` work when `a` is a scalar, but we won't get here in that case becuase the `ValueError` gets raised instead.